docs/controllers/boshdeployment.md

```diff
@@ -0,0 +1,79 @@
+# BOSHDeployment
+
+- [BOSHDeployment](#bosheployment)
+  - [Description](#description)
+  - [Reconcilers](#reconcilers)
+    - [Deployment Reconciler](#deployment-reconciler)
+    - [Generated Variable Reconciler](#generated-variable-reconciler)
+    - [BPM Reconciler](#bpm-reconciler)
+
+## Description
+
+A BOSH deployment is created from a deployment manifest and optionally ops files.
+
+The deployment manifest is based on a vanilla BOSH deployment manifest,
+The ops files modify the deployment manifest. For example, ops files can be used to replace release tarballs with [docker images](https://ci.flintstone.cf.cloud.ibm.com/teams/containerization/pipelines/release-images), thus enabling deployment on Kubernetes.
+
+A deployment is represented by the `boshdeployments.fissile.cloudfoundry.org` (`bdpl`) custom resource, defined in [`boshdeployment_crd.yaml`](https://github.com/cloudfoundry-incubator/cf-operator/tree/master/deploy/helm/cf-operator/templates/fissile_v1alpha1_boshdeployment_crd.yaml).
+This [bdpl custom resource](https://github.com/cloudfoundry-incubator/cf-operator/tree/master/docs/examples/bosh-deployment/boshdeployment.yaml) contains references to config maps or secrets containing the actual manifests content.
+
+After creating the bdpl resource on kubernetes, i.e. via `kubectl apply`, the CF operator will start the reconciliation, which will [eventually result in the deployment](https://docs.google.com/drawings/d/126ExNqPxDg1LcB14pbtS5S-iJzLYPyXZ5Jr9vTfFqXA/edit?usp=sharing) of the BOSH release on Kubernetes.
+
+## Reconcilers
+
+### Deployment Reconciler
+
+Watches for:
+
+- `BOSHDeployment`
+- `ConfigMap`/`Secret` for ops files and the deployment manifest
+- `Pods` created for the BOSHDeployment
```

Comment:
https://github.com/cloudfoundry-incubator/cf-operator/blob/master/pkg/kube/controllers/boshdeployment/deployment_controller.go#L37-L44

But why?

Reply:
Maybe we could clean up this Pod watch if it's OK. I didn't find any Pods belonged to a boshDeployment

Reply:
There's no reason to watch for pods at this level. If we had to, we'd be doing something wrong.

Reply:
anybody working on removing it? Or should I create a story?